project_generator/project.py

```diff
@@ -490,7 +468,8 @@ def generate_dict_for_tool(self, tool):
             'template': toolchain_specific_settings.template or [
                 tool_settings.template for tool_settings in tool_specific_settings if tool_settings.template],
         }
-        self.validate_generated_dic(d)
+        if d['linker_file'] == None and d['output_type'] == 'exe':
```

Comment:
Not sure what you guys think, but a 2 line function that is never re-used was a little funky for me. I still would like to refactor this function somehow - but if we start passing projects arounf instead of this dict, the tool classes will also need refactoring.